Google.Api.Gax.Grpc/GrpcApiDescriptor.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Protobuf.Reflection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Provides metadata about an API. This is expected to be constructed with a single instance
+    /// per API; equality is by simple identity.
```

Comment:
Maybe overriding equals and hashcode explicitly just calling base it's more visible than this comment here. Obviously keep this as well.

I was about to leave a comment about missing equals and hashcode :)

Reply:
Also, just looping the loop but, would it be possible to just want to call some operations with one adapter and some operations with another, and you want to make that explicit by only passing the specific FileDescriptors?

It would be great for workarounds where some operations are not working via gRPC once we support REGAPIC for all.

Granted that descriptors are not per operation so this wouldn't cover all cases.

Reply:
Hmm... not sure about overriding just to return the base values. Let's talk about that more.

In terms of calling some operations with one and some with another: if you want to do that, create two different clients :) I don't think this is a degree of flexibility we should be trying to incorporate, really.